osu.Game/Online/LocalUserStatisticsProvider.cs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+using System.Linq;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Game.Extensions;
+using osu.Game.Online.API;
+using osu.Game.Online.API.Requests;
+using osu.Game.Rulesets;
+using osu.Game.Users;
+
+namespace osu.Game.Online
+{
+    /// <summary>
+    /// A component that keeps track of the latest statistics for the local user.
+    /// </summary>
+    public partial class LocalUserStatisticsProvider : Component
+    {
+        private readonly Bindable<UserStatisticsUpdate> statisticsUpdate = new Bindable<UserStatisticsUpdate>();
+
+        /// <summary>
+        /// A bindable communicating updates to the local user's statistics on any ruleset.
+        /// This does not guarantee the presence of old statistics, as it is invoked on initial population of statistics.
+        /// </summary>
+        public IBindable<UserStatisticsUpdate> StatisticsUpdate => statisticsUpdate;
```

Comment:
This may seem odd at first but it's a comfortable method of communicating user statistics updates to any consuming component without going through the boring method of binding to an event, creating a callback function, and unbinding on disposal.

It's also shown to be directly useful for `DifficultyRecommender` to update its state with any update to any ruleset.

Reply:
I'm pretty sure @bdach's whole point was to not have this, but you've still got it present. Am I missing something.

Reply:
No, it is about the component not tracking changes to the game-wide ruleset bindable. This bindable's main and sole purpose is to let any component be aware if a statistics has changed as a result of user logging out and logging in or submitting a new score. It is more of an event than a bindable but it's written as a bindable to make it easier to use rather than potentially fall in memory leak caused by not unsubscribing from the event.

Reply:
just make it an event if it's not accurate content wise....

Reply:
Sure.

Reply:
https://github.com/ppy/osu/pull/27128/commits/74daf85e489bb5402504729303b0f2dd1fa14e2f